internal/dag/builder.go

```diff
@@ -495,10 +495,17 @@ func (b *Builder) computeHTTPProxy(proxy *projcontour.HTTPProxy) {
 		}
 	}
 
-	if proxy.Spec.TCPProxy != nil && (passthrough || enforceTLS) {
-		if !b.processHTTPProxyTCPProxy(sw, proxy, nil, host) {
+	if proxy.Spec.TCPProxy != nil {
```

Comment:
The logic here is pretty dense. I'll take a crack at cleaning it up after this PR lands.